actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -247,7 +247,14 @@ def handle_positional_args(t, args, options, keys)
                 keys -= options.keys
               end
               keys -= inner_options.keys
-              result.merge!(Hash[keys.zip(args)])
+
+              keys.each { |key|
```

Comment:
@tekin sorry to be a pain, but any chance of you changing this to `do … end` block rather than `{…}`